server/src/remote.rs

```diff
@@ -296,19 +304,236 @@ impl FlightService for FlightServiceHandler {
+
+    Ok(())
+}
+
+/// Save the created model table to the database. This includes creating a tags table for the
+/// model table, adding a new row in the model_table_metadata table, and adding a row to the
+/// model_table_columns table for each field column.
+fn save_model_table_to_database(
+    database_path: PathBuf,
+    model_table_metadata: &NewModelTableMetadata,
+) -> Result<(), rusqlite::Error> {
+    // Create a transaction to ensure the database state is consistent across tables.
+    let mut connection = Connection::open(database_path)?;
+    let transaction = connection.transaction()?;
+
+    // Add a column definition for each tag field in the schema.
+    let mut tag_columns = "".to_owned();
+    let mut tag_indices = model_table_metadata.tag_column_indices.iter().peekable();
+
+    while let Some(index) = tag_indices.next() {
+        // If it is the last column in the query, adding a "," results in an SQL syntax error.
+        let suffix = if tag_indices.peek().is_none() {""} else {", "};
+        let field = model_table_metadata.schema.field(*index as usize);
+
+        tag_columns.push_str(format!("{} TEXT NOT NULL{}", field.name(), suffix).as_str());
+    }
+
+    // Create a table_name_tags SQLite table to save the 54-bit tag hashes when ingesting data.
+    // The query is executed with a formatted string since CREATE TABLE cannot take parameters.
+    transaction.execute(format!("CREATE TABLE {}_tags (hash BLOB PRIMARY KEY, {})",
+        model_table_metadata.name, tag_columns).as_str(), ())?;
+
+    // Add a new row in the model_table_metadata table to persist the model table.
+    transaction.execute(
+        "INSERT INTO model_table_metadata (table_name, schema, timestamp_column_index, tag_column_indices)
+             VALUES (?1, ?2, ?3, ?4)",
+        params![
+            model_table_metadata.name,
+            model_table_metadata.schema.to_json().to_string(),
+            model_table_metadata.timestamp_column_index,
+            model_table_metadata.tag_column_indices
+        ]
+    )?;
+
+    // Add a row for each field column to the model_table_columns table.
+    let mut insert_statement = transaction.prepare(
+        "INSERT INTO model_table_columns (table_name, column_name, column_index)
+             VALUES (?1, ?2, ?3)")?;
+
+    for (index, field) in model_table_metadata.schema.fields().iter().enumerate() {
+        // Only add a row for the field if it is not the timestamp or a tag.
+        let in_tag_indices = model_table_metadata.tag_column_indices.contains(&(index as u8));
+        let is_timestamp = index == model_table_metadata.timestamp_column_index as usize;
+
+        if !is_timestamp && !in_tag_indices {
+            insert_statement.execute(params![model_table_metadata.name, field.name(), index])?;
+        }
+    }
+
+    // Explicitly drop the statement to drop the borrow of "transaction" before the commit.
+    std::mem::drop(insert_statement);
```

Comment:
Very minor, but maybe add `use std::mem` and replace `std::mem::drop()` with `mem::drop()` for consistency with [server/src/models/mod.rs](https://github.com/ModelarData/ModelarDB-RS/blob/master/server/src/models/mod.rs#L26).

Reply:
Fixed.